lib/compress/zstd_lazy.c

```diff
@@ -692,8 +697,14 @@ size_t ZSTD_HcFindBestMatch(
         if ((dictMode != ZSTD_extDict) || matchIndex >= dictLimit) {
             const BYTE* const match = base + matchIndex;
             assert(matchIndex >= dictLimit);   /* ensures this is true if dictMode != ZSTD_extDict */
-            if (match[ml] == ip[ml])   /* potentially better */
-                currentMl = ZSTD_count(ip, match, iLimit);
+            if (dictMode == ZSTD_noDict) {
```

Comment:
Quick question : why is this modification protected by `ZSTD_noDict` ?
Is there a case where it wouldn't work if `dictMode != ZSTD_noDict` ?